Lexer from Arnauld & Module Dependency scope from me

witspirit opened the pull request:
Hi Aman, 

I recently stumbled on your IntelliJBehave plugin and found it very helpful. However, in my project setup I experienced some issues regarding sluggishness and in some cases steps failed to be found.

To overcome these issues I modified your plugin to search for steps in the module containing the story file and the module dependencies. 

In addition we used some story syntax that was not handled properly by your current lexer and parser. I found another github user, arnauld, had already done quite some work in creating a new lexer/parser (not sure, but he seems the lead on the Eclipse JBehave plugin) and merged that in.

If you are still interested in persueing this plugin, I propose you consider pulling in the changes I already aggregated and perhaps you could release a new version in the JetBrains repository.

If, however, you are no longer interested in following up on this plugin, I would like to know whether it would be ok for me to publish a new version in the JetBrains repository with proper attribution.

Kind regards, 

  Bert Van Vlerken (https://github.com/witspirit)

witspirit committed:
Added .gitignore to omit IntelliJBehave.zip (deployable plugin) and out folder (build dir)

witspirit committed:
Merge branch 'improve-parser_by_amauld' of https://github.com/jarosite/IntelliJBehave

And resolved conflicts in:
	.gitignore
	IntelliJBehave.iml
	META-INF/plugin.xml
	src/com/github/kumaraman21/intellijbehave/parser/StoryParser.java
	src/com/github/kumaraman21/intellijbehave/resolver/StepDefinitionIterator.java
	src/com/github/kumaraman21/intellijbehave/resolver/StepPsiReference.java

* Added a pom to download some dependencies
* Fiddled with the IML file... Probably not yet what it should be

Basics are working, using my 'story' project approach.

witspirit committed:
Merge branch 'improve-parser_by_amauld' of https://github.com/jarosite/IntelliJBehave

And resolved conflicts in:
	.gitignore
	IntelliJBehave.iml
	META-INF/plugin.xml
	src/com/github/kumaraman21/intellijbehave/parser/StoryParser.java
	src/com/github/kumaraman21/intellijbehave/resolver/StepDefinitionIterator.java
	src/com/github/kumaraman21/intellijbehave/resolver/StepPsiReference.java

* Added a pom to download some dependencies
* Fiddled with the IML file... Probably not yet what it should be

Basics are working, using my 'story' project approach.

(Some missing files from previous commit...)

witspirit committed:
Alternate module description based on local jars in lib and testlib

witspirit committed:
Modified POM to be aligned with the expectations of the module config file (except for the special JFlex)

* Uses copy-dependencies to get the necessary jars and put them in the appropriate location
* Uses clean to also cleanup the jars it copies there

This way, I don't have to check in the binaries. Which is as it should be.

witspirit committed:
Added testLib/ to .gitignore to ensure you don't add testLib stuff by accident or get distracted by it.

witspirit committed:
* Ensured some debugging statements are hidden (protected by flag)
* Inverted responsibility for the content iteration
* Renamed ProjectUtils to ScanUtils to better reflect responsibility
* Scan Module and Module Dependencies for steps instead of entire Project

atifmansoor commented:
Hi Bert,
First of all, I would like to thank you (+Arnauld+jarosite) for your work and contributions to this project.

Now, can you go ahead and publish this new version (under a new name) to the Plugin Repository?  I believe if you publish under a new name, include the license and proper attribution, we should be "fine".  Correct me if I'm wrong.

Atif

witspirit commented:
Hi Atif,

Thanks for getting back to me.

If you no longer seek to update the plugin, I'd be happy to prepare a
version to publish under a new name, with proper attribution and license.
Thanks for your advice on the matter.

Thanks again for getting this plugin started.

Cheers,

  Bert

On Thu, Mar 14, 2013 at 7:47 PM, Atif Mansoor notifications@github.comwrote:

> Hi Bert,
> First of all, I would like to thank you (+Arnauld+jarosite) for your work
> and contributions to this project.
> 
> Now, can you go ahead and publish this new version (under a new name) to
> the Plugin Repository? I believe if you publish under a new name, include
> the license and proper attribution, we should be "fine". Correct me if I'm
> wrong.
> 
> Atif
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/kumaraman21/IntelliJBehave/pull/24#issuecomment-14921603
> .

atifmansoor commented:
Bert, you have confused me with the original author.  I am not kumaraman21, lol

witspirit commented:
Oops :-) You are right, I confused you with the original author, sorry
about that. Still thanks for your contributions !

Nonetheless, I'll consider doing as you advised.

Kind regards,

  Bert

On Thu, Mar 14, 2013 at 11:51 PM, Atif Mansoor notifications@github.comwrote:

> Bert, you have confused me with the original author. I am not kumaraman21,
> lol
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/kumaraman21/IntelliJBehave/pull/24#issuecomment-14934363
> .

Commit:
Work around JBehave performance issue for invalid input - when the curly braces are unmatched.

Commit:
Fixing shifted highlighting of parameters

Commit:
Fixing some possible NPEs

witspirit committed:
Added ignore for atlassian ide connector.

witspirit committed:
Merge branch 'master' of https://github.com/harley84/IntelliJBehave

witspirit committed:
Updated plugin metadata to prepare for publishing

witspirit committed:
First attempt at README.MD

witspirit committed:
Merge branch 'master' of https://github.com/witspirit/IntelliJBehave

Commit:
Fixing NPE

java.lang.NullPointerException
	at com.github.kumaraman21.intellijbehave.highlighter.StoryLocalizedLexer.tokenType(StoryLocalizedLexer.java:427)
	at com.github.kumaraman21.intellijbehave.highlighter.StoryLocalizedLexer.locateToken(StoryLocalizedLexer.java:168)
	at com.github.kumaraman21.intellijbehave.highlighter.StoryLocalizedLexer.advanceIfRequired(StoryLocalizedLexer.java:114)
	at com.github.kumaraman21.intellijbehave.highlighter.StoryLocalizedLexer.getTokenType(StoryLocalizedLexer.java:80)
	at com.intellij.lang.impl.PsiBuilderImpl.a(PsiBuilderImpl.java:209)
	at com.intellij.lang.impl.PsiBuilderImpl.<init>(PsiBuilderImpl.java:177)
	at com.intellij.lang.impl.PsiBuilderImpl.<init>(PsiBuilderImpl.java:151)
	at com.intellij.lang.impl.PsiBuilderImpl.<init>(PsiBuilderImpl.java:185)
	at com.intellij.lang.impl.PsiBuilderFactoryImpl.createBuilder(PsiBuilderFactoryImpl.java:52)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:62)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:43)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.e(LazyParseableElement.java:165)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:209)
	at com.intellij.psi.impl.source.tree.CompositeElement.countChildren(CompositeElement.java:493)
	at com.intellij.psi.impl.source.tree.CompositeElement.getChildren(CompositeElement.java:438)
	at com.github.kumaraman21.intellijbehave.parser.StoryFileImpl.getStory(StoryFileImpl.java:72)
	at com.github.kumaraman21.intellijbehave.parser.StoryFileImpl.getScenarios(StoryFileImpl.java:62)
	at com.github.kumaraman21.intellijbehave.parser.StoryFileImpl.getSteps(StoryFileImpl.java:52)
	at com.github.kumaraman21.intellijbehave.codeInspector.UnusedStepsInspection$StepUsageFinder.processFile(UnusedStepsInspection.java:122)
	at com.intellij.openapi.vfs.VfsUtilCore$1.visitFileEx(VfsUtilCore.java:189)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:204)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:236)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:236)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:236)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:236)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:236)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:236)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:236)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:236)
	at com.intellij.openapi.vfs.VfsUtilCore.iterateChildrenRecursively(VfsUtilCore.java:184)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.iterateContent(ProjectFileIndexImpl.java:75)
	at com.github.kumaraman21.intellijbehave.codeInspector.UnusedStepsInspection$1.visitElement(UnusedStepsInspection.java:96)
	at com.intellij.psi.JavaElementVisitor.visitAnnotation(JavaElementVisitor.java:344)
	at com.intellij.psi.impl.source.tree.java.PsiAnnotationImpl.accept(PsiAnnotationImpl.java:102)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:412)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:370)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:341)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.access$000(LocalInspectionsPass.java:78)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$2.process(LocalInspectionsPass.java:309)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$2.process(LocalInspectionsPass.java:306)
	at com.intellij.concurrency.JobLauncherImpl$2$1.run(JobLauncherImpl.java:115)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:226)
	at com.intellij.concurrency.JobLauncherImpl$2.process(JobLauncherImpl.java:113)
	at com.intellij.concurrency.JobLauncherImpl$1.run(JobLauncherImpl.java:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
	at java.util.concurrent.FutureTask.run(FutureTask.java:166)
	at com.intellij.concurrency.PrioritizedFutureTask.access$101(PrioritizedFutureTask.java:31)
	at com.intellij.concurrency.PrioritizedFutureTask$1.run(PrioritizedFutureTask.java:70)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1182)
	at com.intellij.concurrency.PrioritizedFutureTask.run(PrioritizedFutureTask.java:97)
	at com.intellij.concurrency.JobImpl.scheduleAndWaitForResults(JobImpl.java:136)
	at com.intellij.concurrency.JobLauncherImpl.a(JobLauncherImpl.java:68)
	at com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:110)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:312)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:218)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.collectInformationWithProgress(LocalInspectionsPass.java:136)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:58)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:62)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1$1.run(PassExecutorService.java:357)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1182)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1.run(PassExecutorService.java:348)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:226)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.a(PassExecutorService.java:345)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:321)
	at com.intellij.concurrency.JobLauncherImpl$3.call(JobLauncherImpl.java:145)
	at com.intellij.concurrency.JobLauncherImpl$3.call(JobLauncherImpl.java:142)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
	at java.util.concurrent.FutureTask.run(FutureTask.java:166)
	at com.intellij.concurrency.PrioritizedFutureTask.access$101(PrioritizedFutureTask.java:31)
	at com.intellij.concurrency.PrioritizedFutureTask$1.run(PrioritizedFutureTask.java:70)
	at com.intellij.concurrency.PrioritizedFutureTask.run(PrioritizedFutureTask.java:113)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603)
	at java.lang.Thread.run(Thread.java:722)

dmalch committed:
Updated versions

dmalch committed:
Compatibility with idea 13

dmalch committed:
Update for project file

dmalch committed:
Speed up

dmalch committed:
Speed up

dmalch committed:
Update version

dmalch committed:
Improved checking of steps

dmalch committed:
Improved checking of steps

dmalch committed:
Support for multiline steps

dmalch committed:
Support for multiline steps

dmalch committed:
Highlight params in multiline steps

Commit:
Fixing out of bounds exception when there are no parameters yet in jbehave annotations

Commit:
Optimizations in UnusedStepsInspection

Commit:
For now turning off Unused Step Declaration inspection by default - can be very heavy on large projects with a lot of stories

Commit:
Updating plugin version to be updated at the repository

harley84 committed:
Guava

harley84 committed:
Project file

harley84 committed:
Merge remote-tracking branch 'origin/pr/3'

Conflicts:
	IntelliJBehave.iml

harley84 committed:
Fixing showing the settings

harley84 committed:
Adding a tooltip to settings page and proper capitalization

harley84 committed:
Fixing an out of bounds exception and getting rid of some obsolete references

dmalch committed:
Merge pull request #1 from witspirit/master

Update fork

dmalch committed:
Merge with master

dmalch committed:
Updated lexer. Now it is able to parse multiline steps.

dmalch committed:
Merge branch 'updated_lexer'

Conflicts:
	IntelliJBehave.iml
	META-INF/plugin.xml
	pom.xml
	src/com/github/kumaraman21/intellijbehave/codeInspector/UnusedStepsInspection.java
	src/com/github/kumaraman21/intellijbehave/highlighter/StoryLocalizedLexer.java
	src/com/github/kumaraman21/intellijbehave/highlighter/StorySyntaxHighlighter.java
	src/com/github/kumaraman21/intellijbehave/parser/StepPsiElement.java
	src/com/github/kumaraman21/intellijbehave/runner/RunStoryAction.java
	src/com/github/kumaraman21/intellijbehave/settings/JBehaveSettingsPanel.java

dmalch committed:
Minor touch up

dmalch committed:
Added icon for stories

harley84 committed:
Updating JBehave core version - fixes #6

harley84 committed:
Some minor memory optimizations

harley84 committed:
Merge remote-tracking branch 'origin/pr/8'

Conflicts:
	src/com/github/kumaraman21/intellijbehave/resolver/StepDefinitionAnnotationConverter.java
	src/com/github/kumaraman21/intellijbehave/resolver/StepDefinitionIterator.java

harley84 committed:
Turning the unused steps inspection off by default and preparing version 1.3

dmalch committed:
Merge pull request #2 from witspirit/master

Update fork

harley84 committed:
Some warnings + fixes #10

harley84 committed:
Supporting IN_EXAMPLES flow. Fixes #9

harley84 committed:
Fixing a possible NPE

harley84 committed:
Better handling the created story file extension and progressing the plugin version

harley84 committed:
Adding self to vendors

dmalch committed:
Merge pull request #3 from witspirit/master

Merge 1.4 version

dmalch committed:
Reference resolution has been reworked

dmalch committed:
Refactoring

dmalch committed:
Refactoring

harley84 committed:
Merge remote-tracking branch 'origin/pr/17'

Conflicts:
	META-INF/plugin.xml

harley84 committed:
Updating references and plugin description

dmalch committed:
Merge pull request #4 from witspirit/master

Sync back

dmalch committed:
Fix for the issue #18 Plugin does not recognise pattern variants

harley84 committed:
Updating version

dmalch committed:
Merge pull request #5 from witspirit/master

dmalch committed:
Fixed issue #20

dmalch committed:
Fixed issue #21

dmalch committed:
Fixed issue #21

dmalch committed:
Added spellchecker

dmalch committed:
Fixed project file

RodrigoQuesadaDev committed:
Added support for Kotlin.

harley84 committed:
Merge pull request #33 from RodrigoQuesadaDev/kotlin

Added support for Kotlin.

RodrigoQuesadaDev committed:
Updated code to support Kotlin 1.0.3.

ulmanA committed:
Update StoryCommenter.java

Fix for issue #31 Comment / Uncomment should be using '!-- ' instead of '!--'

ulmanA committed:
Merge pull request #1 from ulmanA/ulmanA-patch-1

Update StoryCommenter.java

harley84 committed:
Merge pull request #35 from ulmanA/master

Fix for issue #31 Comment / Uncomment should be using '!-- ' instead of '!--'

RodrigoQuesadaDev committed:
Optimized step matching and updated code to support Kotlin 1.0.4.

RodrigoQuesadaDev committed:
Switched build system to Gradle and fixed broken StoryLocalizedLexer_SamplesTest.parseExamples() test.

RodrigoQuesadaDev committed:
ISSUE-42: Fixed NPE occurring in JBehaveUtil class.

JenoDK committed:
Bump versions to latests and adjust where necessary

JenoDK committed:
Add a setting to turn off code completion for story files. resolves #46

JenoDK committed:
Merge pull request #47 from JenoDK/master

Fixes for #40 #42 #46 
Change build system to gradle and up the versions (kotlin and IDE versions)

JenoDK committed:
Update README to include some contribute docs

JenoDK committed:
Avoid ClassCastException - resolves #28

JenoDK committed:
Change to build.gradle and plugin.xml to release a new version

picimako committed:
Feature/#52 upgrade versions and support (#54)

* Upgrade version of IntelliJ, gradle wrapper, and some of the dependencies

* Remove seemingly unused hamcrest dependency

* Replace easytesting libraries with assertj-core

* Replace fest assertions with assertj ones in unit tests

* Migrate application components to Services

* Migrate fileTypeFactory extension point to fileType extension point

* Reformat fileType extension point definition

* Add applicationService definitions to plugin.xml instead of using @Service annotation

* Simplify @State annotation usage on in JBehaveSettings and stop using depracated API

* Add missing param attribute and remove depracated method not called by the platform

* Add icon to StoryRunnerConfigurationType

* Remove most deprecated method calls from RunStoryAction and add StoryRunnerConfigurationType to plugin.xml explicitly as a configurationType extension point

* Change component retireval of JBehaveSettings to service retrieval in StoryCompletionContributor

* Remove JBehaveTemplateLoaderService and relocate Story file template to be automatically loaded by IntelliJ

picimako committed:
#58, #60: Upgrade versions to support IntelliJ versions from 2021.1

picimako committed:
#58, 60: Upgrade dependency versions

valfirst committed:
Fix Gradle Wrapper files (#66)

The proper way to upgrade Gradle Wrapper is to run command like:
./gradlew wrapper --gradle-version=7.1.1 --distribution-type=all

More details: https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper

valfirst committed:
Update badge to point to the plugin in JetBrains Marketplace (#65)

valfirst committed:
Use non-deprecated API

JenoDK committed:
Merge pull request #67 from valfirst/use-non-deprecated-api

Use non-deprecated API

JenoDK committed:
Add gradlew build and test workflow action

picimako committed:
Upgrade intellij gradle plugin to 1.3.0. Set untilBuild to 213.*. Replace a deprecated method call to avoid spamming users with such errors.

valfirst committed:
Favor plain Java 8 over Guava (#68)

picimako committed:
Simplify code by using java 11 language features

JenoDK committed:
Update .github/workflows/main.yml

Co-authored-by: Valery Yatsynovich <valfirst@yandex.ru>

JenoDK committed:
Update .github/workflows/main.yml

Co-authored-by: Valery Yatsynovich <valfirst@yandex.ru>

picimako committed:
Update gradle IJ plugin, gradle and JUnit versions. Add support for IJ 2022.1 eap.

picimako committed:
Add structure view to .story files.

picimako committed:
Upgrade gradle wrapper to 7.4

picimako committed:
Upgrade gradle intellij plugin to 1.5.2

picimako committed:
Use bundled Kotlin

picimako committed:
Fix support of IJ2022.1 in JBehaveStepsIndex

picimako committed:
Update changelog for v1.57

picimako committed:
Add limitation note for structure view.

picimako committed:
Add some run configurations for plugin development.

picimako committed:
Add support for IJ2022.2.

picimako committed:
Updated the kotlin and gradle-intellij-plugin versions.

picimako committed:
Add support for IJ-2022.3

picimako committed:
Remove calls to some deprecated methods.

picimako committed:
Add missing plugin descriptor file for the kotlin plugin.

picimako committed:
Add basic lifecycle, composite and narrative support to the Story structure view. Fix its node hierarchy.

picimako committed:
Update plugin version and changelog

picimako committed:
Update gradle-intellij-plugin to 1.10.0

picimako committed:
Merge pull request #76 from witspirit/support-2022-3

Changes for release 1.59

valfirst committed:
Upgrade to Gradle 7.6

picimako committed:
Merge pull request #77 from valfirst/gradle-7-6

Upgrade to Gradle 7.6

picimako committed:
Update gradle-intellij-plugin to 1.13.1

picimako committed:
Use gradle 7.6 in build.gradle as well.

picimako committed:
Update plugin to support IJ-2023.1.

picimako committed:
Merge pull request #78 from witspirit/v160

v1.60

picimako committed:
Revert gradle-intellij-plugin to 1.12.0

valfirst committed:
Migrate to JBehave 5

picimako committed:
Merge pull request #79 from valfirst/migrate-to-jbehave-5

Migrate to JBehave 5

picimako committed:
Merge pull request #69 from witspirit/Add-gradlew-build-test-action

Add gradlew build and test workflow action

valfirst committed:
Fix build in CI

picimako committed:
Merge pull request #80 from valfirst/fix-build-in-ci

Fix build in CI

valfirst committed:
Upgrade to Gradle `8.1.1`

valfirst committed:
CI: align version of checked out IDEA sources and target IDEA version

picimako committed:
Merge pull request #84 from valfirst/align-version-of-checkouted-idea-sources-to-the-target-idea-version

CI: align version of checked out IDEA sources and target IDEA version

valfirst committed:
Merge branch 'witspirit:master' into upgrade-to-gradle-8-1-1

picimako committed:
Merge pull request #83 from valfirst/upgrade-to-gradle-8-1-1

Upgrade to Gradle `8.1.1`

picimako committed:
Update plugin configuration to be based on the platform plugin template

picimako committed:
Refine the readme

picimako committed:
Update and fix a few tests

picimako committed:
Update since and until builds in plugin.xml

picimako committed:
Make JBehaveStepsIndex light service and fix related tests

picimako committed:
Migrate the UI form designer-based settings page to Java based implementation

picimako committed:
Disable some lexer tests

picimako committed:
Disable some lexer tests

picimako committed:
Remove checkout of intellij-community

picimako committed:
Add plugin verifier configuration

picimako committed:
Merge pull request #85 from witspirit/to-plugin-template

Migrate plugin configuration to intellij platform plugin template

picimako committed:
Fix changelog formatting and start semantic versioning

picimako committed:
Merge pull request #86 from witspirit/versioning-update

Fix changelog formatting and start semantic versioning

picimako committed:
Update plugin configuration and dependency versions

picimako committed:
Remove deprecated API usage

picimako committed:
Use org.apache.commons.lang3.LocaleUtils instead of org.apache.commons.lang.LocaleUtils

picimako committed:
Merge pull request #87 from witspirit/v1.62.0

V1.62.0

picimako committed:
Update plugin configuration and dependencies

picimako committed:
Merge pull request #88 from witspirit/v1.63.0

v1.63.0

picimako committed:
Update plugin version to 1.64.0

picimako committed:
Update dependencies and gradle versions

picimako committed:
Make IJ-2023.1.6 the earliest supported IDE version

picimako committed:
Various minor improvements (missing annotations, unnecessary variable elimination, etc.)

picimako committed:
Improve the unused step declaration inspection, and report the method name instead of the whole method

picimako committed:
Improve the undefined step inspection

picimako committed:
Make KotlinPsiClassesLoader.getPsiClasses() JvmStatic and simplify StepDefinitionIterator

picimako committed:
Remove 'StoryAnnotator'

picimako committed:
Minor improvements in StoryCompletionContributor

picimako committed:
Make the ScanUtils class final

picimako committed:
Remove JUnit 4 dependency and split JUnit 5 and JUnit 3 test into separate gradle tasks

picimako committed:
Mark a local variable as const

picimako committed:
Simplify JBehaveStepReferenceProvider

picimako committed:
Introduce caching in JBehaveStepsIndex

picimako committed:
Add Gradle test task include-exclude configuration

picimako committed:
Simplify KotlinAnnotationsLoader a little

picimako committed:
Minor adjustments in JBehaveUtil

picimako committed:
Cover JBehaveUtil with tests, and improve the code in it

picimako committed:
Add a null check when querying the Ref object in the PSI change listener

picimako committed:
Add tests for JBehaveJavaStepDefinitionSearch and improve its implementation

picimako committed:
Increase test coverage of JBehaveUtil. Add a short-circuit condition when searching for the biggest word in blank strings.

picimako committed:
Add tests for JBehaveJavaMethodUsageSearcher. Deduplicate ReferencesSearch query calls in it.

picimako committed:
Add tests and javadoc for JavaStepDefinition. Optimize its implementation by minimizing the number of OptimizedStepMatchers instantiated.

picimako committed:
Add tests for StepPsiReference and StepPsiReferenceContributor. Improve the implementation of StepPsiReference

picimako committed:
Remove an unused import

picimako committed:
Move JBehaveStepsIndex tests to JUnit5 implementation

picimako committed:
Update plugin to support IJ-2024.2-EAP.

picimako committed:
Merge pull request #89 from witspirit/v1.64.0

V1.64.0

picimako committed:
Add caching for the JBehave step reference resolution and Given-When-Then step annotation lookup.

picimako committed:
Rename a few variables, and remove an unnecessary variable creation

picimako committed:
Implement a simplified and optimized version of JBehave's PatternVariantBuilder

picimako committed:
Use already available parent method, if available, for annotations in JBehaveUtil.getAnnotationTexts()

picimako committed:
Add some collection related fine-tunings in PatternVariantBuilder

picimako committed:
Deduplicate annotation type retrieval, and StepPatternParser static, in JavaStepDefinition.

picimako committed:
Lower the number of search scope creations in JBehaveJavaStepDefinitionSearch

picimako committed:
Cache step definition finding results per JBehaveStep in JBehaveStepsIndex

picimako committed:
Add licence NOTICE

picimako committed:
Merge pull request #90 from witspirit/1.65.0

1.65.0

picimako committed:
Add code documentation to a handful of Kotlin step definitions related classes

picimako committed:
Updated IDE support range to 2023.2.8 - 2024.3-EAP

picimako committed:
Update dependencies, GHA workflow and gradle wrapper

picimako committed:
New supported IDE version range: 2023.2.8 - 2024.2.0.2

picimako committed:
Merge pull request #91 from witspirit/1660

v1.66.0

picimako committed:
Update project to IJ Platform Gradle Plugin 2.0 and JDK 21

picimako committed:
Fix JBehaveStepDefClassPsiChangeListener to no longer fail when it encounters invalid files

picimako committed:
Make sure the plugin works with the Kotlin K2 compiler

picimako committed:
Fix a potential exception during highlighting undefined steps

picimako committed:
Remove a couple of deprecated API usage

picimako committed:
Update JUnit3 test run configuration to use the correct gradle task

picimako committed:
Fix JUnit3 gradle task name in GHA workflow

picimako committed:
Merge pull request #92 from witspirit/1670

v1.67.0

picimako committed:
Use the proper name for the kotlin K2 compiler extension

picimako committed:
Remove unnecessary since-until build versions from the plugin.xml

picimako committed:
Update the Run Verifications run configuration to use the task from the platform plugin v2.0

picimako committed:
Merge pull request #93 from witspirit/1671

v1.67.1

picimako committed:
Minor performance improvements in PatternVariantBuilder

picimako committed:
#94: Improve matching steps with patterns with empty parameter values

picimako committed:
Add some javadoc in JBehaveStep

picimako committed:
Update changelog

picimako committed:
Merge pull request #95 from witspirit/1672

v1.67.2

picimako committed:
Update class name in license attributions

picimako committed:
Update gradle wrapper to 8.10.2

picimako committed:
Update dependency versions

picimako committed:
Update GitHub Actions workflow configuration

picimako committed:
Add exception handling when analyzing Kotlin classes if they are step definition classes

picimako committed:
Merge pull request #96 from witspirit/1673

Add exception handling when analyzing Kotlin classes if they are step definition classes

picimako committed:
Add exception handling when analyzing Java classes if they are step definition classes

picimako committed:
Fix code comment in JBehaveStepDefClassPsiChangeListener

picimako committed:
Merge pull request #97 from witspirit/1674

v1.67.4

picimako committed:
Add further exception handling when analyzing Java and Kotlin classes if they are step definition classes.

picimako committed:
Fix changelog sections

picimako committed:
Merge pull request #98 from witspirit/1675

v1.67.5

picimako committed:
Update support version range to 2024.3-2025.1-EAP, and update necessary configuration along with it.

picimako committed:
Merge pull request #99 from witspirit/1680

v1.68.0

Commit:
Fixed a bug where references were not being resolved for valid lines.

This appear to be due to the cache key not being provided and therefore defaulting to the name of the anonymous lambda

picimako committed:
Update dependencies and plugin configuration

picimako committed:
Unname exception variables

picimako committed:
Fix some tests

picimako committed:
Reorganize dependencies and replace test execution with the simple 'test' task

picimako committed:
Use ReadAction.compute() with static import

picimako committed:
Add changelog entry

picimako committed:
Delete unused run configurations

picimako committed:
Delete unused imports

picimako committed:
Replace lambda with method reference

picimako committed:
Merge pull request #100 from tomb50/fix/caching-key

Step reference not being resolved for valid lines

picimako committed:
Merge branch 'master' into 1690

# Conflicts:
#	src/main/java/com/github/kumaraman21/intellijbehave/service/JBehaveStepsIndex.java

picimako committed:
Add bugfix section to changelog

picimako committed:
Merge pull request #101 from witspirit/1690

v1.69.0

picimako committed:
Exclude JUnit Jupiter dependencies from the plugin archive

picimako committed:
Merge pull request #102 from witspirit/1691

Exclude JUnit Jupiter dependencies from the plugin archive

picimako committed:
Introduce MergingUpdateQueue in JBehaveStepDefClassPsiChangeListener to minimize running update logic during PSI changes

picimako committed:
Reduce the number of Stream creations in JBehaveUtil

picimako committed:
Move some ReadAction.compute() calls to JBehaveUtil-.isStepDefinition

picimako committed:
Reduce the number of project content iterations regarding unused step analysis

picimako committed:
Add a todo comment to StepAnnotationsCache for future cache cleanup

picimako committed:
Simplify, deduplicate, reduce the number of ReadActions in StepDefinitionAnnotationConverter and StepDefinitionIterator. Increase test coverage.

picimako committed:
Retrieve only methods of the current class and only public ones when identifying step def methods

picimako committed:
Update plugin version to 1.69.2

picimako committed:
Update intellij-platform version to 2.11.0

picimako committed:
Merge pull request #104 from witspirit/1692

v1.69.2
